server/src/test/java/org/elasticsearch/action/admin/cluster/allocation/TransportGetDesiredBalanceActionTests.java

```diff
@@ -267,14 +278,14 @@ public void testGetDesiredBalance() throws Exception {
                     );
                     assertEquals(indexMetadata.getTierPreference(), shardView.tierPreference());
                 }
-                Optional<ShardAssignment> shardAssignment = Optional.ofNullable(shardAssignments.get(indexShardRoutingTable.shardId()));
-                if (shardAssignment.isPresent()) {
-                    assertEquals(shardAssignment.get().nodeIds(), desiredShard.desired().nodeIds());
-                    assertEquals(shardAssignment.get().total(), desiredShard.desired().total());
-                    assertEquals(shardAssignment.get().unassigned(), desiredShard.desired().unassigned());
-                    assertEquals(shardAssignment.get().ignored(), desiredShard.desired().ignored());
+                final var shardAssignment = shardAssignments.get(indexShardRoutingTable.shardId());
+                if (shardAssignment == null) {
```

Comment:
Is this test generating null assignments?

Reply:
Yes it only assigns some random subset of the shards.